nerfstudio/data/datamanagers/base_datamanager.py

```diff
@@ -86,14 +90,38 @@ def variable_res_collate(batch: List[Dict]) -> Dict:
         # now that iteration is complete, the image data items can be removed from the batch
         for key in topop:
             del data[key]
-
     new_batch = nerfstudio_collate(batch)
     new_batch["image"] = images
     new_batch.update(imgdata_lists)
 
     return new_batch
 
 
+def ray_collate(batch: List[RayBundle]):
+    # start = time.time()
+    ray_bundle_list, batch_list = list(zip(*batch))
+    combined_metadata = {}
+    if "fisheye_crop_radius" in ray_bundle_list[0].metadata:
+        combined_metadata["fisheye_crop_radius"] = ray_bundle_list[0].metadata["fisheye_crop_radius"]
+    if "directions_norm" in ray_bundle_list[0].metadata:
+        combined_metadata["directions_norm"] = torch.cat([ray_bundle_i.metadata["directions_norm"] for ray_bundle_i in ray_bundle_list], dim=0)
+
+    concatenated_ray_bundle = RayBundle(
+        origins=torch.cat([ray_bundle_i.origins for ray_bundle_i in ray_bundle_list], dim=0),
+        directions=torch.cat([ray_bundle_i.directions for ray_bundle_i in ray_bundle_list], dim=0),
+        pixel_area=torch.cat([ray_bundle_i.pixel_area for ray_bundle_i in ray_bundle_list], dim=0),
+        camera_indices=torch.cat([ray_bundle_i.camera_indices for ray_bundle_i in ray_bundle_list], dim=0),
+        metadata=combined_metadata,
+    )
+    concatenated_batch = {
+        "image" : torch.cat([batch_i["image"] for batch_i in batch_list], dim=0),
+        "indices": torch.cat([batch_i["indices"] for batch_i in batch_list], dim=0),
+    }
+    # end = time.time()
+    # print((end - start) * 1000)
```

Comment:
curious, this function is pretty fast right?

Reply:
Actually it's not that fast, it takes around 0.24 seconds per raybundle which adds a lot of time overall - I've removed this because of that